cli/clianalysisreport.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Cppcheck - A tool for static C/C++ code analysis
+ * Copyright (C) 2007-2022 Cppcheck team.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "clianalysisreport.h"
+
+CLIAnalysisReport::CLIAnalysisReport(bool verbose, std::string templateFormat, std::string templateLocation)
+    : mVerbose(verbose), mTemplateFormat(std::move(templateFormat)), mTemplateLocation(std::move(templateLocation)) {}
+
+void CLIAnalysisReport::addFinding(const ErrorMessage &msg) {
+    mBuffer << msg.toString(mVerbose, mTemplateFormat, mTemplateLocation);
```

Comment:
I get the feeling that errors are not reported directly with this code?

Reply:
I would like that CLI warnings are reported directly. And that warnings in the GUI are shown directly. But for SARIF warnings I feel it's ok that the report is generated at the end.

Reply:
That makes sense. I've deleted some of my changes to the CLI errors/warnings. So, the only changes in the PR now are related to XML and SARIF output. All of the other output should be the same.